internal/service/alarm_mapper.go

```diff
@@ -0,0 +1,332 @@
+	logger        *slog.Logger
+	backendClient *http.Client
+	//transportWrapper    func(http.RoundTripper) http.RoundTripper
+	resourceServerURL   string
+	resourceServerToken string
+	extensions          []string
+}
+
+// AlarmMapper creates a builder that can then be used to configure
+// and create a handler for the AlarmMapper.
+func NewAlarmMapper() *AlarmMapperBuilder {
+	return &AlarmMapperBuilder{}
+}
+
+// SetLogger sets the logger that the handler will use to write to the log. This is mandatory.
+func (b *AlarmMapperBuilder) SetLogger(
+	value *slog.Logger) *AlarmMapperBuilder {
+	b.logger = value
+	return b
+}
+
+// SetBackendClient sets the backend client used to connect to other servers. This is mandatory.
+func (b *AlarmMapperBuilder) SetBackendClient(value *http.Client) *AlarmMapperBuilder {
+	b.backendClient = value
+	return b
+}
+
+// SetResourceServerURL sets the URL of the resource server. This is mandatory.
+// The resource server is used for mapping Alarms to Resources.
+func (b *AlarmMapperBuilder) SetResourceServerURL(
+	value string) *AlarmMapperBuilder {
+	b.resourceServerURL = value
+	return b
+}
+
+// SetResourceServerToken sets the authentication token that will be used to authenticate
+// with to the resource server. This is mandatory.
+func (b *AlarmMapperBuilder) SetResourceServerToken(
+	value string) *AlarmMapperBuilder {
+	b.resourceServerToken = value
+	return b
+}
+
+// SetExtensions sets the fields that will be added to the extensions. This is optional.
+func (b *AlarmMapperBuilder) SetExtensions(values ...string) *AlarmMapperBuilder {
+	b.extensions = values
+	return b
+}
+
+// Build uses the data stored in the builder to create and configure a new handler.
+func (b *AlarmMapperBuilder) Build() (
+	result *AlarmMapper, err error) {
+	// Check parameters:
+	if b.logger == nil {
+		err = errors.New("logger is mandatory")
+		return
+	}
+	if b.backendClient == nil {
+		err = errors.New("backend client is mandatory")
+		return
+	}
+
+	// Create a jq compiler function for parsing labels
+	compilerFunc := gojq.WithFunction("parse_labels", 0, 1, func(x any, _ []any) any {
+		if labels, ok := x.(string); ok {
+			return data.GetLabelsMap(labels)
+		}
+		return nil
+	})
+
+	// Create the jq tool:
+	jqTool, err := jq.NewTool().
+		SetLogger(b.logger).
+		SetCompilerOption(&compilerFunc).
+		Build()
+	if err != nil {
+		return
+	}
+
+	// Check that extensions are at least syntactically valid:
+	for _, extension := range b.extensions {
+		_, err = jqTool.Compile(extension)
+		if err != nil {
+			return
+		}
+	}
+
+	// Create and populate the object:
+	result = &AlarmMapper{
+		logger:              b.logger,
+		backendClient:       b.backendClient,
+		resourceServerURL:   b.resourceServerURL,
+		resourceServerToken: b.resourceServerToken,
+		extensions:          b.extensions,
+		jqTool:              jqTool,
+	}
+	return
+}
+
+// Map an Alert to an O2 Alarm object.
+func (r *AlarmMapper) MapAlertItem(ctx context.Context,
```

Comment:
As this is an alert mapper there is no need to repeat it in the name of the method. I'd suggest just `Map`. Alternatively you could just return the function from the builder, and then there is no need for another name.

Reply:
good point.